### DISCUSSION TITLE
iCub faceExpressions custom RFE BM_M4 failed: "command not recognized"

### DISCUSSION THREAD

**Discussion by lunmre** [robotology/community#694] [Q&A] [answered]
Hi there,
I'm trying to add a new custom facial expression on iCub, specially I need to custom the LEDs of the RFE board. I followed the instruction here: https://robotology.github.io/robotology-documentation/doc/html/group__icub__faceExpressions.html
Things going well when I tried the 'set mask' command with almost all the pre-defined bitmask:
`set mask (white BM_EB1 5) (white BM_EB1 5) (white BM_M0 5)`
except for this `BM_M4` with:
`set mask (white BM_EB1 5) (white BM_EB1 5) (white BM_M4 5)`
I got: Response: [fail] "command not recognized"

Since I don't know which `emotions_rfe.ini` the iCub robot is actually using, I checked all the `emotions_rfe.ini` files I can find, both in the `icub-head` and on the `icubsrv`.

![Screenshot from 2024-03-29 14-11-37](https://github.com/robotology/community/assets/15604075/e86443e0-e718-4f2c-a1c8-72ad295926bd)

![Screenshot from 2024-03-29 14-15-55](https://github.com/robotology/community/assets/15604075/4ec1d3f4-ebbe-4fa1-8ffa-0f9fc5725266)

![Screenshot from 2024-03-29 14-25-55](https://github.com/robotology/community/assets/15604075/5a161e73-1261-431f-909d-8c5a97fd1f7a)

In all these `emotions_rfe.ini` configs, the 'BM_M4' seems to be set correctly. So 
it confuses me what makes the `BM_M4` different from other custom bitmasks such as `BM_M3`?

Also if I want to add a new custom bitmask such as `BM_M5`, which `emotions_rfe.ini` shall I modify?

**Comment by Nicogene** [comment, marked as answer]
Hi @lunmre!

The error you are encountering is due to wrong-index bug:
- https://github.com/robotology/icub-main/pull/955

With that should be working. 

> Also if I want to add a new custom bitmask such as BM_M5, which emotions_rfe.ini shall I modify?

You can, you must increase `bitmask_mouth_emotions` to **6** add the new bitmask in your emotions_rfe.ini.

For reference of what those numbers means see this figure

![immagine](https://github.com/robotology/community/assets/19152494/d5bd59e5-fca8-4b6c-9c47-f8a7f8c132a0)

**Reply by lunmre** [reply]
Great! I'll give it a try. Much appreciated!
